### PR TITLE
AUTH-296: crypto: add a CA{} method for creating subCAs, too

### PR DESCRIPTION
**Pull request by stlaz**
While working on microshift, I noticed that this method was missing.

/assign @deads2k 
/cc @ibihim

### CODE REVIEW

**Comment by deads2k** [comment]
Required I assume?  The PR looks reasonable to me

/approve
/assign @ibihim 

@ibihim owns lgtm

**Comment by stlaz** [comment]
> Required I assume? The PR looks reasonable to me

It's going to make our life in Microshift a bit easier, it makes sense to include this method from the perspective of the code I added there.

**Review by ibihim — APPROVED** [review]
Let me check

File: pkg/crypto/crypto.go [hunk 1]
```diff
@@ -693,6 +693,43 @@ func MakeCAConfigForDuration(name string, caLifetime time.Duration, issuer *CA)
 	return signerConfig, nil
 }
 
+func (ca *CA) EnsureSubCA(certFile, keyFile, serialFile, name string, expireDays int) (*CA, bool, error) {
```
Comment (ibihim) [inline pkg/crypto/crypto.go:696]: Nit: It would be nice to have a comment on an exported method.

File: pkg/crypto/crypto.go [hunk 2]
```diff
@@ -693,6 +693,43 @@ func MakeCAConfigForDuration(name string, caLifetime time.Duration, issuer *CA)
...
+	if subCA, err := GetCA(certFile, keyFile, serialFile); err == nil {
+		return subCA, false, err
+	}
+	subCA, err := ca.MakeAndWriteSubCA(certFile, keyFile, serialFile, name, expireDays)
+	return subCA, true, err
+}
+
+func (ca *CA) MakeAndWriteSubCA(certFile, keyFile, serialFile, name string, expireDays int) (*CA, error) {
```
Comment (ibihim) [inline pkg/crypto/crypto.go:704]: Nit: It would be nice to have a comment on an exported method.

File: pkg/crypto/crypto.go [hunk 3]
```diff
@@ -693,6 +693,43 @@ func MakeCAConfigForDuration(name string, caLifetime time.Duration, issuer *CA)
 	return signerConfig, nil
 }
 
+func (ca *CA) EnsureSubCA(certFile, keyFile, serialFile, name string, expireDays int) (*CA, bool, error) {
+	if subCA, err := GetCA(certFile, keyFile, serialFile); err == nil {
+		return subCA, false, err
```
Comment (ibihim) [inline pkg/crypto/crypto.go:698]: What is the point of the boolean value?

**Pull request merged**